exchange/binance.go

```diff
@@ -529,7 +577,26 @@ func (b *Binance) CandlesByPeriod(ctx context.Context, pair, period string,
 	}
 
 	for _, d := range data {
-		candles = append(candles, CandleFromKline(pair, *d))
+		candle := CandleFromKline(pair, *d)
+
+		if b.CandleType == "heikinAshi" {
+			haCandle := ha.CalculateHeikinAshi(candle)
```

Comment:
Maybe, we can move this call `CalculateHeikinAshi` to `CandleFromKline` and receive a candle type as parameter. It will avoid duplication in line 541.

Reply:
Thanks for your feedback! I made a few changes to the code. Basically, I removed the `HeikinAshi` model and I use the existing `Candle` model instead. The code seems cleaner now because we avoid duplication, and we can transform a default candle to another type (heikin ashi in our case) by using the following method. 
```go
   candle = candle.ToHeikinAshi(ha)
```

Let me know what you think. 🙏🏻